kernel/src/scan/mod.rs

```diff
@@ -441,6 +450,133 @@ impl Scan {
     pub fn scan_metadata(
         &self,
         engine: &dyn Engine,
+    ) -> DeltaResult<impl Iterator<Item = DeltaResult<ScanMetadata>>> {
+        self.scan_metadata_inner(engine, self.replay_for_scan_metadata(engine)?)
+    }
+
+    /// Get an updated iterator of [`ScanMetadata`]s based on an existing iterator of [`ScanMetadata`]s.
+    ///
+    /// The existing iterator is assumed to have originated from a previous call to  `scan_metadata`.
```

Comment:
can we explicitly call out that this expects a 'materialized' `scan_metadata` output? that is you have to apply the selection vector?